### ISSUE TITLE
No-arg DefaultConfiguration() constructor clones default settings twice

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14960]
This issue's discussion showed that there's a lot of potential performance improvement when avoiding creating new `DefaultConfiguration` instances, and cloning its internal `Settings`:
https://github.com/jOOQ/jOOQ/issues/6683#issuecomment-1514691496

While the thorough fix is delicate, there's a very low hanging fruit that can be fixed without any risk. When calling `new DefaultConfiguration()` (which jOOQ's internals do a lot, see #6683), we clone the internal default `Settings` twice in a row:

```
java.lang.Throwable
	at org.jooq.conf.SettingsBase.clone(SettingsBase.java:59)
	at org.jooq.conf.Settings.clone(Settings.java:1)
	at org.jooq.conf.SettingsTools.clone(SettingsTools.java:375)
	at org.jooq.conf.SettingsTools.defaultSettings(SettingsTools.java:368)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:174)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:164)
	at org.jooq.testscripts.JDBC.main(JDBC.java:45)
java.lang.Throwable
	at org.jooq.conf.SettingsBase.clone(SettingsBase.java:59)
	at org.jooq.conf.Settings.clone(Settings.java:1)
	at org.jooq.conf.SettingsTools.clone(SettingsTools.java:375)
	at org.jooq.impl.DefaultConfiguration.set(DefaultConfiguration.java:1522)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:327)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:212)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:174)
	at org.jooq.impl.DefaultConfiguration.<init>(DefaultConfiguration.java:164)
	at org.jooq.testscripts.JDBC.main(JDBC.java:45)
```

This is completely unnecessary and can be avoided very easily.

> Note that cloning is currently necessary because both `DefaultConfiguration` and `Settings` are mutable, and we don't want to risk anyone accidentally mutating the defaults. See #6683 for details.

----
See also:
- https://github.com/jOOQ/jOOQ/issues/6683#issuecomment-1514691496
- https://github.com/jOOQ/jOOQ/issues/14959

**Comment by lukaseder** [comment]
A quick rendering benchmark that makes internal use of `DSL.select()` and other similar calls produces a 3% improvement:

**2 clone calls**

```
ValuesOfRecordsBenchmark.valuesOfRecords  thrpt    7  1017.603 ┬▒ 14.404  ops/s
```

**1 clone call**

```
ValuesOfRecordsBenchmark.valuesOfRecords  thrpt    7  1050.685 ┬▒ 3.837  ops/s
```

As always, it's just a benchmark, not representative of real world querying, which may or may not require the internal creation of such subqueries.

Details here: https://github.com/jOOQ/jOOQ/issues/6683

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.4 (https://github.com/jOOQ/jOOQ/issues/14961), 3.17.13 (https://github.com/jOOQ/jOOQ/issues/14962), 3.16.19 (https://github.com/jOOQ/jOOQ/issues/14963)
